runtime/lua/vim/lsp/linked_editing_range.lua

```diff
@@ -0,0 +1,199 @@
+local ns = api.nvim_create_namespace('nvim.lsp.linked_editing_range')
+
+--- |lsp-handler| for the method `textDocument/linkedEditingRange`
+--- Stores ranges globally, since they only make sense to track for the current buffer
+---@param result lsp.LinkedEditingRanges
+---@param ctx lsp.HandlerContext
+---@private
+function M.on_linkededitingrange(err, result, ctx)
+  if err then
+    log.error('linkededitingrange', err)
+    return
+  end
+  local bufnr = assert(ctx.bufnr)
+  api.nvim_buf_clear_namespace(0, ns, 0, -1)
+  if
+    util.buf_versions[bufnr] ~= ctx.version
+    or not result
+    or not api.nvim_buf_is_loaded(bufnr)
+    or not globalstate.enabled
+  then
+    return
+  end
+
+  local client_id = ctx.client_id
+  local client = assert(vim.lsp.get_client_by_id(client_id))
+
+  local lines = api.nvim_buf_get_lines(bufnr, 0, -1, false)
+  local curpos = api.nvim_win_get_cursor(0)
+  local cursor_range = { curpos[1] - 1, curpos[2], curpos[1] - 1, curpos[2] }
+  for i, range in ipairs(result.ranges) do
+    local start_line = range.start.line
+    local line = lines and lines[start_line + 1] or ''
+    range.start.character =
+      vim.str_byteindex(line, client.offset_encoding, range.start.character, false)
+    local end_line = range['end'].line
+    line = lines and lines[end_line + 1] or ''
+    range['end'].character =
+      vim.str_byteindex(line, client.offset_encoding, range['end'].character, false)
+
+    api.nvim_buf_set_extmark(bufnr, ns, start_line, range.start.character, {
+      end_line = end_line,
+      end_col = range['end'].character,
+      hl_group = 'LspLinkedEditingRange',
+      right_gravity = false,
+      end_right_gravity = true,
+    })
+
+    local range_tuple =
+      { range.start.line, range.start.character, range['end'].line, range['end'].character }
+    if Range.contains(range_tuple, cursor_range) then
+      globalstate.range_index = i
+    end
+  end
+
+  if not globalstate.range_index then
+    globalstate.enabled = false
+  end
+  -- TODO: Use the word pattern to determine if range content is valid
+  globalstate.word_pattern = result.wordPattern
+end
+
+--- Refresh linked editing ranges, only if we have attached clients that support it
+---@param bufnr (integer) Buffer handle, or 0 for current
+---@param opts? vim.lsp.util._refresh.Opts Additional options to pass to util._refresh
+---@private
+local function _refresh(bufnr, opts)
+  opts = opts or {}
+  local win = api.nvim_get_current_win()
+  opts['bufnr'] = bufnr
+  local method = ms.textDocument_linkedEditingRange
+  local clients = vim.lsp.get_clients({ bufnr = bufnr, method = method })
+
+  if #clients == 0 then
+    return
+  end
+  -- TODO: Merge results from multiple clients?
+  local client = clients[1]
+  client:request(
+    method,
+    vim.lsp.util.make_position_params(win, client.offset_encoding),
+    opts.handler,
+    bufnr
+  )
+end
+
+--- Enable linked editing ranges for a buffer
+local function _enable()
```

Comment:
Nit: Remove the `_` from local functions. We use that prefix to skip docs for module methods, but we don't need to do that for local functions.